app/controllers/interaction_claims_controller.rb

```diff
@@ -55,4 +51,40 @@ def perform_interaction_search
     search_results = LookupInteractions.find(params)
     @search_results = InteractionSearchResultsPresenter.new(search_results, view_context)
   end
+
+  def unpack_locals params
+    @preset_fda = (params[:fda_approved_drug] == "checked" ? "FDA Approved" : "")
+    @preset_neo = (params[:anti_neoplastic] == "checked" ? "Anti-neoplastics" : "")
+    @preset_immuno = (params[:immunotherapy] == "checked" ? "Immunotherapies" : "")
+    @preset_clin = (params[:clinically_actionable] == "checked" ? "Clinically Actionable" : "")
+    @preset_druggable = (params[:druggable_genome] == "checked" ? "Druggable Genome" : "")
+    @preset_resist = (params[:drug_resistance] == "checked" ? "Drug Resistance" : "")
+  end
+
+  def prepare_export(compound_field_separator = '|')
+    headers = %w[search_term match_term match_type gene drug interaction_types sources pmids]
+    @tsv = CSV.generate(col_sep: "\t") do |tsv|
+      tsv << headers
+      @search_results.instance_variable_get(:@search_results).each do |result|
+        match_field = result.instance_variable_get(:@type).singularize + '_id'
+        row_hash = {
+            'match_type' => result.match_type_label,
+            'search_term' => result.instance_variable_get(:@search_term)
+        }
+        identifiers = result.instance_variable_get(:@identifiers)
+        # result.instance_variable_get(:@identifiers).each do |id|
+        #   row_hash['match_term'] = id.name
+        result.instance_variable_get(:@interactions).each do |interaction|
```

Comment:
Yeah, I agree with what you're saying--the controller felt like a weird place to put this. Is there a way that I could pull this logic out to a presenter, but retain the data I need (the search results) without performing a second search?